### PR TITLE
RxJava ConcurrentSyntax & Dispatchers

### PR DESCRIPTION
**Pull request by nomisRev**
I was investigating #1776  but could not reproduce.

This PR enables `ConcurrentSyntax` syntax for `fx` instead of `AsyncSyntax`.
To do so it adds `Dispatchers` instances for all RxJava types and exposes a `(Scheduler) -> CoroutineContext` function.

### CODE REVIEW

File: modules/fx/arrow-fx-rx2/src/test/kotlin/arrow/fx/SingleKTests.kt [hunk 1]
```diff
@@ -66,50 +69,61 @@ class SingleKTests : RxJavaSpec() {
...
     "Multi-thread Singles finish correctly" {
-      val value: Single<Long> = SingleK.fx {
-        val a = Single.timer(2, TimeUnit.SECONDS).k().bind()
-        a
-      }.value()
-
-      val test: TestObserver<Long> = value.test()
-      test.awaitDone(5, TimeUnit.SECONDS)
-      test.assertTerminated().assertComplete().assertNoErrors().assertValue(0)
+      forFew(10, Gen.choose(10L, 50)) { delay ->
```
Comment (nomisRev) [inline modules/fx/arrow-fx-rx2/src/test/kotlin/arrow/fx/SingleKTests.kt]: Could not reproduce even when increased to 1000 tries.

File: modules/fx/arrow-fx-rx2/src/test/kotlin/arrow/fx/SingleKTests.kt [hunk 2]
```diff
@@ -66,50 +69,61 @@ class SingleKTests : RxJavaSpec() {
...
+        SingleK.fx {
+          val a = Single.timer(delay, TimeUnit.MILLISECONDS).k().bind()
```
Comment (aballano) [inline modules/fx/arrow-fx-rx2/src/test/kotlin/arrow/fx/SingleKTests.kt]: Subtle but very nice change, I tested locally and got around 16% lower compile time (from ~4:20 to ~3:40) 🎉

File: modules/fx/arrow-fx-rx2/src/test/kotlin/arrow/fx/SingleKTests.kt (hunk 1 above)
Comment (aballano) [reply modules/fx/arrow-fx-rx2/src/test/kotlin/arrow/fx/SingleKTests.kt]: I think we could close #1776 and reopen if we see this happening again after your changes 👍

**Comment by nomisRev** [comment]
Closes #1776

File: modules/core/arrow-test/src/main/kotlin/arrow/test/generators/Generators.kt [hunk 3]
```diff
@@ -145,3 +145,6 @@ fun Gen.Companion.char(): Gen<Char> =
   Gen.from(('A'..'Z') + ('a'..'z') + ('0'..'9') + "!@#$%%^&*()_-~`,<.?/:;}{][±§".toList())
 
 fun <A> Gen.Companion.genSetK(genA: Gen<A>): Gen<SetK<A>> = Gen.set(genA).map { it.k() }
+
+fun Gen.Companion.unit(): Gen<Unit> =
```
Comment (raulraja) [inline modules/core/arrow-test/src/main/kotlin/arrow/test/generators/Generators.kt]: We should consider porting these to kotlintest arrrow module

**Pull request merged**